Jellyfin.Api/Controllers/LibraryController.cs

```diff
@@ -640,7 +643,47 @@ public async Task<ActionResult> GetDownload([FromRoute, Required] Guid itemId)
             // Quotes are valid in linux. They'll possibly cause issues here.
             var filename = Path.GetFileName(item.Path)?.Replace("\"", string.Empty, StringComparison.Ordinal);
 
-            return PhysicalFile(item.Path, MimeTypes.GetMimeType(item.Path), filename, true);
+            var streams = item.GetMediaStreams();
+            var uniqueFiles = new List<string>();
+            foreach (var stream in streams)
+            {
+                if (!(string.IsNullOrEmpty(stream.Path) || uniqueFiles.Contains(stream.Path)))
+                {
+                    uniqueFiles.Add(stream.Path);
+                }
+            }
+
+            uniqueFiles.Add(item.Path);
+
+            if (uniqueFiles.Count > 1)
```

Comment:
This is a breaking change. It should at least be optional.

Reply:
Well, currently there is only a single download action. Making this optional is more difficult, as we would need to make it configureable somehow, i.e. by adding new parameters to the download endpoint. This would then require changes to clients to actually send these parameters which also needs new UI probably?

I found the current download option to be not really useful, since the current UI makes downloads of subtitles almost impossible (you have to talk to a SubtitleController endpoint directly, there's no UI). This change at least makes it behave it in a way I would expect from a download point of view. The current download logic already returned arbitrary filetypes (depending on what was stored on disk), so it's only semi-breaking IMHO.

If you have a suggestion on how to make this configurable, preferably without new UI, I'm happy to add switches.

Reply:
Why is a UI change such a bad thing? A simple query parameter like `?includeExternalStreams=true` would make the intention clear and without breaking anything